app-frontend/src/app/components/tools/diagramContainer/diagramContainer.controller.js

```diff
@@ -51,29 +51,141 @@ export default class DiagramContainerController {
-                    data-menu-options="menuOptions"
                   ></rf-diagram-node-header>
+                  <div class="node-actions">
+                    <div class="node-button-group">
+                      <button class="btn node-button" type="button"
+                              ng-if="onPreview"
+                              ng-click="onPreview($event, this.model)">
+                      <span class="icon-eye"></span>
+                      </button>
+                      <button class="btn node-button" type="button"
+                              ng-class="{'active': showHistogram}"
+                              ng-click="toggleHistogram()">
+                      <span class="icon-histogram"></span>
+                      </button>
+                      <button class="btn node-button" type="button"
+                              ng-click="toggleBody()">
+                      <span ng-class="{'icon-caret-up': showBody,
+                                       'icon-caret-down': !showBody}"></span>
+                      </button>
+                    </div>
+                  </div>
                   <rf-input-node
                     ng-if="model.get('cellType') === 'src'"
+                    ng-show="showBody && !showHistogram"
                     data-model="model"
                     on-change="onChange({sourceId: sourceId, project: project, band: band})"
                   ></rf-input-node>
                   <rf-operation-node
                     ng-if="model.get('cellType') === 'function'"
+                    ng-show="showBody && !showHistogram"
                     data-model="model"
                   ></rf-operation-node>
                   <rf-constant-node
                     ng-if="model.get('cellType') === 'const'"
+                    ng-show="showBody && !showHistogram"
                     data-model="model"
                     on-change="onChange({override: override})"
                   ></rf-constant-node>
+                  <rf-node-histogram
+                    ng-if="showHistogram && showBody"
+                    data-histogram="histogram"
+                    data-breakpoints="breakpoints"
+                    data-options="histogramOptions"
+                    data-masks="masks"
+                    on-breakpoint-change="onBreakpointChange(breakpoints, masks)"
+                  ></rf-node-histogram>
                 </div>`,
             initialize: function () {
                 _.bindAll(this, 'updateBox');
                 joint.dia.ElementView.prototype.initialize.apply(this, arguments);
                 this.model.on('change', this.updateBox, this);
                 this.$box = angular.element(this.template);
                 this.scope = $scope.$new();
+                this.scope.showBody = true;
+                this.scope.breakpoints = {
+                    1: { value: 0, style: 'bar', alwaysShowNumbers: true, color: '#ff0000' },
```

Comment:
This doesn't work well for negative values. If I create a tool with the definition `A*_G` and then set G=-1 in the Lab interface, these values will still get posted to the API as positives, which will cause every tile to display as red (since they are all lower than 0). This can be seen in a less artificial context by viewing the Subtraction node of NDVI; the default ramp will look fine, but after applying these defaults it will turn red wherever the subtraction ends up less than zero.

Reply:
the current color scheme (and values) are placeholders until alex finishes #2467. I think when we integrate the two pieces, we can examine this further.